### PR TITLE
Fix error popups when class extends itself

### LINKED ISSUE #919: Error popup if a type extends itself
eg. try creating the following class in Eclipse with javac

```java
public class HelloWorld extends HelloWorld {

}
```

There should be an error popup that you can't get rid of once you've saved this file

### PR DESCRIPTION
**Pull request by datho7561**
Mostly null checks in bindings, plus override type translation to prevent trying to access the broken class.

Fixes #919

### CODE REVIEW

File: org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/ProceedOnErrorTransTypes.java [hunk 1]
```diff
@@ -41,5 +42,12 @@ public void visitApply(JCMethodInvocation tree) {
 		}
 		super.visitApply(tree);
 	}
+	
+	@Override
+	public void visitIdent(JCIdent tree) {
+		if (tree.type.getEnclosingType() == null) {
+			return;
+		}
```
Comment (mickaelistria) [inline org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/ProceedOnErrorTransTypes.java:67]: missing call to `super.visitIdent(tree)` in the regular case.

**Review by mickaelistria — CHANGES_REQUESTED** [review]

File: org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/ProceedOnErrorTransTypes.java (hunk 1 above)
Comment (datho7561) [reply org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/ProceedOnErrorTransTypes.java:67]: :facepalm:

**Comment by datho7561** [comment]
The TransType stuff I wrote is completely wrong, I'll need to take a proper look at this.

**Comment by mickaelistria** [comment]
I think this can be merged. Can you please rebase and push?

**Comment by mickaelistria** [comment]
Those change do cause a lot of build errors for regular non erroneous cases. I think it's best to revert at the moment.

**Pull request merged**